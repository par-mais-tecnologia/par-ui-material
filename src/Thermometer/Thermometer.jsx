import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { importAllImages } from '../Core/functions'

const images = importAllImages(require.context('./assets', false, /\.(svg)$/))

class Thermometer extends PureComponent {
  render () {
    const { bioColor } = this.props
    const img = images[`BIO_${bioColor.toUpperCase()}_MOBILE.svg`]
    return (
      <img className='t'
        src={img}
      />
    )
  }
}

Thermometer.propTypes = {
  bioColor: PropTypes.string
}

Thermometer.defaultProps = {
  bioColor: 'azul'
}

export default Thermometer
