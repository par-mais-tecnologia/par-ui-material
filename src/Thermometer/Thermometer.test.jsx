import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Thermometer from './'

configure({ adapter: new Adapter() })

export const test = describe('Thermometer', () => {
  it('shallow Thermometer', () => {
    const wrapper = shallow(
      <Thermometer />
    )
    expect(wrapper).toBeTruthy()
  })

  it('shallow Thermometer match Snapshot', () => {
    const wrapper = shallow(
      <Thermometer />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
