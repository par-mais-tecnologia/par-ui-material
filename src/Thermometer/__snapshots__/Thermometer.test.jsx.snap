// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thermometer shallow Thermometer match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Thermometer
    bioColor="azul"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "t",
      "src": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "t",
        "src": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
